### PR TITLE
SCRD-3501 Added delete compute page as part of replacing a compute server

### PR DESCRIPTION
**Pull request by ghost**
This commit includes:
1. Added delete compute page.
2. Refactored the replace detail page to use ConfirmModal to have
  consistent look and feel as the rest of the dialogs.
3. Renamed  osInstallPassword to be osPassword and make it  available
  without osInstall for replacing a compute.
4. Updated to handle empty IPMI values when select and deselect
installOS for compute server.
5. Always show osUsername and osPassword inputs for compute server
6. Updated to have installOS page when user chooses to installOS
7. Moved some methods related to logging progress to misc utils
8. Fixed inconsistent colon usage in the compelete page

### CODE REVIEW

**Comment by guglo** [comment]
The change still has to go with https://gerrit.suse.provo.cloud//#/c/5300/

**Comment by jeremy-moffitt** [comment]
I'm a little confused by `Added delete compute page.` ... Is this as part of the replace workflow? Or is it supposed to be a standalone action? I don't see a "delete" action anywhere on the computes list

**Comment by jeremy-moffitt** [comment]
@guglo thanks for clarifying, based on the original commit message I thought there might be a whole new page for deleting servers (via the menu in the server summary page or similar), I understand now that this is specific to the replace workflow.

**Comment by jeremy-moffitt** [comment]
tried it out with the mock backend, works great there, peeking through the code now

**Review by jeremy-moffitt — CHANGES_REQUESTED** [review]
mostly looks good... couple of questions, including one that I'm pretty sure is a typo that needs to be fixed

File: src/pages/ReplaceServer/DeleteCompute.js [hunk 1]
```diff
@@ -0,0 +1,564 @@
...
+import { ConfirmModal } from '../../components/Modals.js';
+import { logProgressResponse, logProgressError } from '../../utils/MiscUtils.js';
+import {
+  NOVA_STOP_PLAYBOOK, NEUTRON_STOP_PLAYBOOK, BM_POWER_DOWN_PLAYBOOK,
+  COBBLER_DEPLOY_PLAYBOOK, PRE_DEPLOYMENT_PLAYBOOK, MONASCA_DEPLOY_PLAYBOOK,
+  MODEL_SERVER_PROPS_ALL
+} from '../../utils/constants.js';
+import { removeServerFromModel, getMergedServer, genUID } from '../../utils/ModelUtils.js';
+
+const MANUAL_SHUTDOWN = 'manual_shutdonw';
```
[the remark this answers is not in the document]
Comment (guglo) [reply src/pages/ReplaceServer/DeleteCompute.js]: oh...will fix

File: src/pages/ReplaceServer/DeleteCompute.js [hunk 2]
```diff
@@ -0,0 +1,564 @@
...
+  }
+
+  removeOldServer = (resolve, reject, logger) => {
+    let server =
+      this.props.model.getIn(['inputModel','servers'])
+        .find(server => server.get('id') ===  this.props.operationProps.oldServer.id);
+    if (server) {
+      server = server.toJS();
+      let model =
+        removeServerFromModel(server, this.props.model);
```
[the remark this answers is not in the document]
Comment (guglo) [reply src/pages/ReplaceServer/DeleteCompute.js]: not sure why...one line should be fine. will update

File: src/pages/ReplaceServer/DeleteCompute.js [hunk 3]
```diff
@@ -0,0 +1,564 @@
...
+              // if cannot update the deleted server info in the saved servers
+              // for some reason, still let it go
+              resolve();
+            });
+        }
+        else { // default server in the model, but not in the saved servers, will add server
+          if(!server['uid']) {
+            server['uid'] = genUID('manual');
+          }
+          if(!server['manual']) {
```
[the remark this answers is not in the document]
Comment (guglo) [reply src/pages/ReplaceServer/DeleteCompute.js]: should be !server['source']...will fix.

**Review by jeremy-moffitt — APPROVED** [review]
LGTM, worked in the dev environment

**Review by GarySmith — APPROVED** [review]
Looks good, but I noticed a few cleanup items are necessary. I filed SCRD-6374, SCRD-6375, and SCRD-6376 to cover those.

**Pull request merged**